flang/runtime/ragged.cpp

```diff
@@ -11,28 +11,21 @@
 
 namespace Fortran::runtime {
 
-inline bool isIndirection(const RaggedArrayHeader *const header) {
-  return header->flags & 1;
-}
-
-inline std::size_t rank(const RaggedArrayHeader *const header) {
-  return header->flags >> 1;
-}
-
 RaggedArrayHeader *RaggedArrayAllocate(RaggedArrayHeader *header, bool isHeader,
     std::int64_t rank, std::int64_t elementSize, std::int64_t *extentVector) {
   if (header && rank) {
-    std::int64_t size = 1;
+    std::int64_t size{1};
     for (std::int64_t counter{0}; counter < rank; ++counter) {
       size *= extentVector[counter];
       if (size <= 0) {
         return nullptr;
       }
     }
-    header->flags = (rank << 1) | isHeader;
+    header->indirection = isHeader;
+    header->rank = rank;
     header->extentPointer = extentVector;
     if (isHeader) {
-      header->bufferPointer = new RaggedArrayHeader[size];
+      header->bufferPointer = std::malloc(sizeof(RaggedArrayHeader) * size);
```

Comment:
malloc does not initialize the memory returned, right? So how does this guarantee the block is set to all 0s? This looks like a bug.

Reply:
This was also a change requested by Peter.